### ISSUE TITLE
SUBTASK 4 Training Integration

### ISSUE THREAD

**Issue by shanjiaz** [vllm-project/speculators#331]
Originally a task in [Add P-eagle support in training](https://github.com/vllm-project/speculators/issues/292). 

## Task
- Add cross entropy loss function as an option in training script
- Group-specific hidden state and target extraction at sampled indices
- Per-depth accuracy metrics
- Training loop modifications for single-pass parallel group processing
